pypesto/objective/aesara.py

```diff
@@ -90,6 +91,41 @@ def __init__(self,
         # temporary storage for evaluation results of objective
         self.inner_ret: ResultDict = {}
 
+    def get_negloglikelihood(
+        self,
+        x: np.ndarray,
+        sensi_orders: Tuple[int, ...] = (0, ),
+        mode: str = MODE_FUN,
+        return_dict: bool = False,
+        **kwargs,
+    ) -> Union[float, np.ndarray, Tuple, ResultDict]:
+        """
+        Negative loglikelihood.
+
+        Returns the negative loglikelihood, which in this case is the
+        objective itself.
+        """
+        return super().__call__(x, sensi_orders, mode, return_dict, **kwargs)
+
+    def get_neglogprior(
+        self,
+        x: np.ndarray,
+        sensi_orders: Tuple[int, ...] = (0, ),
+        mode: str = MODE_FUN,
+        return_dict: bool = False,
+        **kwargs,
+    ) -> Union[float, np.ndarray, Tuple, ResultDict]:
+        """
+        Negative logprior.
+
+        As the Objective is interpreted as negative loglikelihood, the prior
+        is the "trivial" prior.
+        """
+        if return_dict:
+            get_zero_result_dict(x, sensi_orders, mode)
```

Comment:
the base objective may have a prior, so I think this function should compute input to the base function and then call `get_neglogprior` on the base function. Same for likelihood. Might be easiest to implement this through an additional argument that can be passed to `__call__` to avoid code duplication.

Reply:
good point. :+1: Had a closer look again, and just calling the corresponding methods on `self.base_objective` should resolve that issue.

Reply:
but `self.base_objective` needs `self.infun(x)` as input, not `x`. Also `inner_ret` needs to be populated. Happy to provide an implementation.

Reply:
Thank you for your offer, that would be great! :)

Reply:
done

Reply:
Thank you :)